hudi-spark-datasource/hudi-spark-common/src/main/java/org/apache/hudi/DataSourceUtils.java

```diff
@@ -193,7 +194,8 @@ public static HoodieWriteConfig createHoodieConfig(String schemaStr, String base
         .withCompactionConfig(HoodieCompactionConfig.newBuilder()
             .withInlineCompaction(inlineCompact).build())
         .withPayloadConfig(HoodiePayloadConfig.newBuilder()
-            .withPayloadClass(parameters.get(DataSourceWriteOptions.PAYLOAD_CLASS_NAME().key()))
+            .withPayloadClass(parameters.getOrDefault(DataSourceWriteOptions.PAYLOAD_CLASS_NAME().key(),
```

Comment:
Have you checked all such places that use `PAYLOAD_CLASS_NAME`, so they should also get the payload class name from the table config?

Reply:
Expression payload is not from the table config so we can't just grab it from the table config. We do testing to make sure custom payloads are respected

Reply:
Got it.  Let's a comment in the code to mention this nuance.